### PR TITLE
Implement vsomeip-sys to wrap vsomeip C++ library

### PR DESCRIPTION
**Pull request by PLeVasseur**
Implements #11

Breaking this off as PR number 1/2 from #2

To Dos

- [x] Add CI

Opened an issue for adding unit and integration tests

### CODE REVIEW

File: .github/workflows/lint-and-test.yaml [hunk 1]
```diff
@@ -28,16 +40,171 @@ concurrency:
...
+          key: ${{ env.CACHE_KEY }}
+          restore-keys: |
+            ${{ runner.os }}-vsomeip-
+
+      - name: Use vsomeip installation
+        run: |
+          echo "VSOMEIP install path from previous job: ${{ env.VSOMEIP_INSTALL_PATH }}"
+          ls -l ${{ env.VSOMEIP_INSTALL_PATH }}
+
+# TODO: Temporarily commenting this proven bit out to get the download & compilation
```
Comment (devkelley) [inline .github/workflows/lint-and-test.yaml:161]: Can be removed now?

File: .github/workflows/lint-and-test.yaml [hunk 2]
```diff
@@ -28,16 +40,171 @@ concurrency:
...
+          cmake --build build --target install
+
+      - name: Save vsomeip build to cache
+        if: steps.cache-vsomeip.outputs.cache-hit != 'true'
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.VSOMEIP_INSTALL_PATH }}
+          key: ${{ runner.os }}-vsomeip-${{ hashFiles(env.VSOMEIP_SOURCE_PATH_WITH_WILDCARD) }}
+
+  foo:
```
Comment (devkelley) [inline .github/workflows/lint-and-test.yaml:122]: maybe a more descriptive job name?

File: Cargo.toml [hunk 3]
```diff
@@ -25,10 +25,10 @@ exclude = [
 ]
 keywords = ["uProtocol", "SDK", "vsomeip", "SOMEIP"]
 license = "Apache-2.0"
-name = "up-client-vsomeip-rust"
+name = "up-transport-vsomeip"
 readme = "README.md"
 repository = "https://github.com/eclipse-uprotocol/up-client-vsomeip-rust"
-rust-version = "1.72"
+rust-version = "1.74"
```
Comment (devkelley) [inline Cargo.toml:31]: why 1.74 rather than 1.76? To keep parity with the up-rust library?

File: .github/workflows/lint-and-test.yaml (hunk 2 above)
Comment (PLeVasseur) [reply .github/workflows/lint-and-test.yaml:122]: Hah, sorry. I'm still in the midst of updating the CI. Feel free to review everything but that :)

File: Cargo.toml (hunk 3 above)
Comment (PLeVasseur) [reply Cargo.toml:36]: up-transport-zenoh-rust uses 1.74

File: .github/workflows/lint-and-test.yaml (hunk 1 above)
Comment (PLeVasseur) [reply .github/workflows/lint-and-test.yaml:161]: Again... haha, feel free to review everything but the CI :)

File: .github/workflows/lint-and-test.yaml (hunk 2 above)
Comment (devkelley) [reply .github/workflows/lint-and-test.yaml:122]: fair enough :)

File: Cargo.toml (hunk 3 above)
Comment (devkelley) [reply Cargo.toml:36]: Got it, and what was the reasoning behind removing the toolchain file?

Comment (PLeVasseur) [reply Cargo.toml:36]: cargo tarpaulin was failing at 1.74 and I need to work through why. Until then, I just removed the toolchain file :upside_down_face:

Comment (devkelley) [reply Cargo.toml:36]: Ah got it! FWIW, I had a similar issue with tarpaulin and it resolved itself once I upgraded to 1.76. Has something to do with tarpaulin using a feature not available until that version

Comment (PLeVasseur) [reply Cargo.toml:36]: Gotcha. Zenoh will upgrade to 1.75 for their 1.0.0 release, so I was hoping it'd just resolve then. But maybe not hah

**Comment by PLeVasseur** [comment]
Okay @devkelley & all: the CI is now updated to work!

We can see it run a single unit test in vsomeip-sys I had written in the past :)

https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/actions/runs/9786728167/job/27022105149

**Comment by PLeVasseur** [comment]
Taking off of draft status now that CI works! :slightly_smiling_face:

File: .github/workflows/lint-and-test.yaml [hunk 4]
```diff
@@ -13,6 +13,18 @@
 
 name: Lint and Test
 
+env:
+  VSOMEIP_RELEASE_URL: https://github.com/COVESA/vsomeip/archive/refs/tags/
+  VSOMEIP_RELEASE_TAG: 3.4.10.tar.gz
+  VSOMEIP_SOURCE_PATH: vsomeip-src
+  VSOMEIP_INSTALL_PATH: vsomeip-install
+
+  # TODO: Need to get a cached version of vsomeip compiled we can refer to here
```
Comment (PLeVasseur) [inline .github/workflows/lint-and-test.yaml:22]: We do have a cached version and the variable used for this is `VSOMEIP_INSTALL_PATH` from above. Should delete this TODO.

File: .github/workflows/lint-and-test.yaml [hunk 5]
```diff
@@ -13,6 +13,18 @@
...
+  VSOMEIP_LIB_PATH: /does/not/yet/exist
+  # TODO: Need to get a C++ build environment setup with C++ libs in known locations
```
Comment (PLeVasseur) [inline .github/workflows/lint-and-test.yaml:24]: This is also done, should remove TODO

File: vsomeip-sys/build.rs [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+        File::create(&temp_file_path).expect("Failed to create a temporary file for writing");
+    let mut writer = BufWriter::new(temp_file);
+
+    for line in reader.lines() {
+        let mut line = line.expect("Failed to read a line from the input file");
+        line = line.replace(
+            "pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete",
+            "#[allow(unused_imports)] pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete"
+        );
+        line = line.replace(
```
Comment (stevenhartley) [inline vsomeip-sys/build.rs:141]: Do we not want to create a patch and apply it like what Ivan was doing for the vsomeip helloworld example? Actually looking over the code again it looks like you are overriding the rust generated code is that correct? Why?

File: vsomeip-sys/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+use autocxx::prelude::*; // use all the main autocxx functions
+
+// using autocxx to generate Rust bindings for all these
+include_cpp! {
+    #include "vsomeip/vsomeip.hpp"
+    #include "runtime_wrapper.h"
+    #include "application_wrapper.h"
+    #include "message_wrapper.h"
+    #include "payload_wrapper.h"
+    safety!(unsafe) // see details of unsafety policies described in the 'safety' section of the book
```
Comment (stevenhartley) [inline vsomeip-sys/src/lib.rs:25]: unsafe? Why?

**Review by stevenhartley — COMMENTED** [review]
95% of the code is over my head and hard to follow. I hope @sophokles73 and @devkelley can help cover there. I wonder if we should put vsomeip-sys Inside of COVESA like we did for vsomeip-python project, what do you think?

File: vsomeip-sys/build.rs (hunk 6 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs:141]: Yes, overriding the generated Rust code in a few spots to:
1. eliminate warnings
2. add the ability to output enum fields

File: vsomeip-sys/src/lib.rs (hunk 7 above)
Comment (PLeVasseur) [reply vsomeip-sys/src/lib.rs:25]: We're wrapping C++ code, which seems to mean unsafe by default.

**Comment by PLeVasseur** [comment]
> 95% of the code is over my head and hard to follow. I hope @sophokles73 and @devkelley can help cover there. I wonder if we should put vsomeip-sys Inside of COVESA like we did for vsomeip-python project, what do you think?

From what I saw of vsomeip-python it looked like a more complete wrapper. vsomeip-sys is, for the moment, just enough to enable building up-transport-vsomeip-rust.

If we have the time / inclination to cover a broader area of vsomeip and a larger maintenance load, it's possible to upstream.

File: vsomeip-sys/README.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+## Supplying your own vsomeip includes
+
+You may modify the build command above to include setting the environment variable:
+
+```
+VSOMEIP_INCLUDE_PATH=<path/to/vsomeip/include>
+```
+
+`VSOMEIP_INCLUDE_PATH` must point to the folder which has the `vsomeip` folder within it containing the vsomeip includes.
```
Comment (sophokles73) [inline vsomeip-sys/README.md:45]: My understanding is that we _always_ require the user to have the vsomeip shared library installed, i.e. we do not create it as part of the vsomeip-sys build.
If that is the case, then I would suggest to also require the user to _always_ provide the (matching) header files. Otherwise we might run into situations where the user provides the lib in version A and we download the source code in version B. This doesn't seem to pose any additional effort for the user because he needs to build vsomeip from source anyway, as there is no binary distribution and as part of the build/install process the vsomeip headers will also be installed, right?

File: vsomeip-sys/build.rs [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+            let vsomeip_source_folder = Path::new(&out_dir).join("vsomeip").join("vsomeip-src");
+            download_and_write_file(&vsomeip_archive_url, &vsomeip_archive_dest)
+                .expect("Unable to download released archive");
+            decompress::decompress(
+                vsomeip_archive_dest,
+                vsomeip_source_folder.clone(),
+                &ExtractOptsBuilder::default().strip(1).build().unwrap(),
+            )
+            .expect("Unable to extract tar.gz");
+            vsomeip_source_folder.join("interface")
```
Comment (sophokles73) [inline vsomeip-sys/build.rs:49]: FMPOV we should rely on the user always providing the path

File: vsomeip-sys/build.rs [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+    let arch_specific_cpp_stdlib = env::var("ARCH_SPECIFIC_CPP_STDLIB_PATH").expect("You must supply the path to architecture-specific C++ stdlib, e.g. /usr/include/x86_64-linux-gnu/c++/11");
+
+    let project_root = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    let runtime_wrapper_dir = project_root.join("src/glue/include"); // Update the path as necessary
+
+    // Somewhat useful debugging
+    // println!("cargo:warning=# CARGO_MANIFEST_DIR : {}", project_root.display());
+    // println!("cargo:warning=# OUT_DIR            : {}", out_path.display());
+    // println!("cargo:warning=# vsomeip_interface  : {}", interface_path.display());
+    // println!("cargo:warning=# runtime_wrapper    : {}", runtime_wrapper_dir.display());
```
Comment (sophokles73) [inline vsomeip-sys/build.rs:66]: if this is really useful then why is it commented out?

File: vsomeip-sys/build.rs [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+        .include(&vsomeip_interface_path)
+        .include(&runtime_wrapper_dir)
+        .flag_if_supported("-Wno-deprecated-declarations") // suppress warnings from C++
+        .flag_if_supported("-Wno-unused-function") // compiler compiling vsomeip
+        .flag_if_supported("-std=c++17")
+        .extra_warnings(true)
+        .compile("cxx-portion");
+    println!("cargo:rerun-if-changed=src/cxx_bridge.rs");
+
+    // we rewrite the autocxx generated code to suppress the cargo warning about unused imports
```
Comment (sophokles73) [inline vsomeip-sys/build.rs:104]: why do we need to suppress these warnings? Do they break the build?

File: vsomeip-sys/build.rs [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+        );
+        line = line.replace("pub unsafe fn create_payload1", "unsafe fn create_payload1");
+        line = line.replace("successfully [de]registered", "successfully de/registered");
+        line = line.replace(
+            "#[derive(Clone, Hash, PartialEq, Eq)]",
+            "#[derive(Clone, Hash, PartialEq, Eq, Debug)]",
+        );
+        line = line.replace(
+            "pub unsafe fn set_data(self: Pin<&mut payload>, _data: *const u8, _length: u32);",
+            "pub(crate) unsafe fn set_data(self: Pin<&mut payload>, _data: *const u8, _length: u32);"
```
Comment (sophokles73) [inline vsomeip-sys/build.rs:157]: why do you need to reduce visibility here?

File: vsomeip-sys/build.rs [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+    println!("cargo:warning=# rewrote the autocxx file");
+
+    Ok(())
+}
+
+// Retrieves a file from `url` (from GitHub, for instance) and places it in the build directory (`OUT_DIR`) with the name
+// provided by `destination` parameter.
+fn download_and_write_file(url: &str, dest_path: &PathBuf) -> Result<(), Box<dyn Error>> {
+    let client = Client::builder()
+        .timeout(Duration::from_secs(120)) // Set a timeout of 60 seconds
```
Comment (sophokles73) [inline vsomeip-sys/build.rs:211]: looks like a timeout of 120 secs to me, anyway, the comment seems superfluous to me ;-)

File: vsomeip-sys/src/lib.rs [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+    generate!("glue::get_payload_raw")
+    generate!("glue::create_payload_wrapper")
+}
+
+pub mod vsomeip {
+    pub use crate::ffi::vsomeip_v3::*;
+}
+
+pub mod extern_callback_wrappers;
+pub mod safe_glue;
```
Comment (sophokles73) [inline vsomeip-sys/src/lib.rs:57]: having a public API in a module called `glue` and `safe_glue` feels a little odd. Do you really need to expose these modules in their entirety? Maybe you can simple reexport some types of rename the modules?

File: vsomeip-sys/src/safe_glue.rs [hunk 15]
```diff
@@ -0,0 +1,455 @@
...
+/// Because vsomeip makes heavy use of std::shared_ptr and Rust's ownership and borrowing model
+/// doesn't work well together with it, in Rust code use a UniquePtr<[RuntimeWrapper]>
+///
+/// Since we use a UniquePtr<[RuntimeWrapper]>, we then need a way to drill down and extract
+/// the `Pin<&mut runtime>`.
+///
+/// # TODO
+///
+/// Add some runtime safety checks on the pointer
+pub fn get_pinned_runtime(wrapper: &RuntimeWrapper) -> Pin<&mut runtime> {
```
Comment (sophokles73) [inline vsomeip-sys/src/safe_glue.rs:47]: can these explicit functions maybe replaced with some clever implementation of `Deref` instead?

**Review by sophokles73 — COMMENTED** [review]
I have deliberately skipped reviewing any of the c++ files since I have no clue about c++ ;-) So this is as far as I can support ...

File: vsomeip-sys/build.rs (hunk 9 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs:49]: Yeah we chatted about this. I'll probably have to make this change for the reasons you outlined above.

File: vsomeip-sys/build.rs (hunk 10 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs:66]: Good point. I need to find a better way that doesn't show up as a yellow, scary warning :sweat_smile:

File: vsomeip-sys/build.rs (hunk 11 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs:104]: They break the build since we fail on warnings emitted from the generated Rust code

File: vsomeip-sys/src/lib.rs (hunk 14 above)
Comment (PLeVasseur) [reply vsomeip-sys/src/lib.rs:57]: Generally speaking... I should probably revisit the module setup.

I agree with you. I may open an issue for this instead of addressing on this PR since it'll take a bit of rearrangement.

File: vsomeip-sys/src/safe_glue.rs (hunk 15 above)
Comment (PLeVasseur) [reply vsomeip-sys/src/safe_glue.rs:49]: I guess this is not allowed since the borrow checker will not allow for pinned mutable borrows to be returned from Deref.

I spent some time on this today, but didn't see a way around the issue.

Darn. I'm going to call this one unsolvable for the moment :(

File: .github/workflows/lint-and-test.yaml (hunk 1 above)
Comment (PLeVasseur) [reply .github/workflows/lint-and-test.yaml]: Fixed

File: .github/workflows/lint-and-test.yaml (hunk 2 above)
Comment (PLeVasseur) [reply .github/workflows/lint-and-test.yaml]: foo removed :)

File: vsomeip-sys/src/safe_glue.rs (hunk 15 above)
Comment (PLeVasseur) [reply vsomeip-sys/src/safe_glue.rs]: I updated this to clarify and hopefully simplify the API.

File: up-transport-vsomeip/Cargo.toml [hunk 16]
```diff
@@ -0,0 +1,39 @@
+################################################################################
+# Copyright (c) 2023 Contributors to the Eclipse Foundation
```
[the remark this answers is not in the document]
Comment (PLeVasseur) [reply up-transport-vsomeip/Cargo.toml]: Done

File: vsomeip-sys/Cargo.toml [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+lazy_static = "1.4.0"
+once_cell = "1.19.0"
+log = "0.4.21"
+
+[build-dependencies]
+autocxx-build = { version = "0.26.0" }
+cxx-build = "1.0.122"
+decompress = { version = "0.6.0" }
+miette = { version = "5", features = ["fancy"] }
+reqwest = { version = "0.12", features = ["blocking"] }
```
[the remark this answers is not in the document]
Comment (PLeVasseur) [reply vsomeip-sys/Cargo.toml]: Done

File: vsomeip-sys/build.rs (hunk 10 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs]: I couldn't find a way to make this a non-scary yellow warning, but I find it to be helpful regardless

File: vsomeip-sys/build.rs (hunk 11 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs:113]: Updated to clarify what's going on by breaking apart

File: vsomeip-sys/build.rs (hunk 12 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs]: Broke these apart to clarify

File: vsomeip-sys/build.rs (hunk 13 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs]: Resolved :laughing:

File: vsomeip-sys/README.md (hunk 8 above)
Comment (PLeVasseur) [reply vsomeip-sys/README.md]: Added the ability to do this in the latest commit. Looking for feedback

File: vsomeip-sys/build.rs (hunk 9 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs]: Added the ability to compile the downloaded vsomeip library for the user in the latest commit. Looking for feedback :slightly_smiling_face:

File: vsomeip-sys/build.rs [hunk 18]
```diff
@@ -134,81 +143,72 @@ fn main() -> miette::Result<()> {
...
-
     println!("cargo:warning=# rewrote the autocxx file");
 
     Ok(())
 }
 
+fn fix_unused_imports(line: String) -> String {
+    let mut fixed_line = line.replace(
+        "pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete",
+        "#[allow(unused_imports)] pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete",
```
Comment (sophokles73) [inline vsomeip-sys/build.rs:228]: Am I right in assuming that the generated code itself does not need the type but that the vsomeip transport will need it?

File: vsomeip-sys/build.rs [hunk 19]
```diff
@@ -134,81 +143,72 @@ fn main() -> miette::Result<()> {
...
-            "pub(crate) unsafe fn set_data(self: Pin<&mut payload>, _data: *const u8, _length: u32);"
-        );
-        // line = line.replace(
-        //     "# [repr (u8)] # [derive (Clone , Hash , PartialEq , Eq)] pub enum message_type_e",
-        //     "# [repr (u8)] # [derive (Clone , Hash , PartialEq , Eq, Debug)] pub enum message_type_e"
-        // );
+        line = fix_unused_imports(line);
+        line = fix_unsafe_fn_unused(line);
+        line = fix_doc_build(line);
+        line = add_enum_debug(line);
```
Comment (sophokles73) [inline vsomeip-sys/build.rs:212]: this is much better :+1:

**Review by sophokles73 — COMMENTED** [review]
I like the idea of being able to build the vsomeip library as part of the transport's build. Have you taken a look at other *-sys crates and how they do this? Maybe https://github.com/eclipse/paho.mqtt.rust/tree/master/paho-mqtt-sys can provide some additional inspiration ...

File: vsomeip-sys/build.rs [hunk 20]
```diff
@@ -60,10 +60,19 @@ fn main() -> miette::Result<()> {
     let runtime_wrapper_dir = project_root.join("src/glue/include"); // Update the path as necessary
 
     // Somewhat useful debugging
-    // println!("cargo:warning=# CARGO_MANIFEST_DIR : {}", project_root.display());
-    // println!("cargo:warning=# OUT_DIR            : {}", out_path.display());
-    // println!("cargo:warning=# vsomeip_interface  : {}", interface_path.display());
-    // println!("cargo:warning=# runtime_wrapper    : {}", runtime_wrapper_dir.display());
+    println!("cargo:warning=# vsomeip_lib_path  : {}", vsomeip_lib_path);
```
[the remark this answers is not in the document]
Comment (PLeVasseur) [reply vsomeip-sys/build.rs:104]: Lemme look into this

File: vsomeip-sys/build.rs (hunk 18 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs:228]: I think we don't use it in the transport. Would you recommend removal entirely then?

Comment (sophokles73) [reply vsomeip-sys/build.rs:228]: Unless you want to create a generic vsomeip-sys crate that is independent of vsomeip-utransport ...

Comment (PLeVasseur) [reply vsomeip-sys/build.rs:228]: No thanks, too much work atm :melting_face: 

Will remove

File: vsomeip-sys/build.rs (hunk 20 above)
[the remark this answers is not in the document]
Comment (PLeVasseur) [reply vsomeip-sys/build.rs:104]: Jammed pack full of lots of other debug stuff if we do this, but it's still more in the spirit of being a debug message, so I'll do it.

[the remark this answers is not in the document]
Comment (PLeVasseur) [reply vsomeip-sys/build.rs]: Done

File: vsomeip-sys/build.rs (hunk 18 above)
Comment (PLeVasseur) [reply vsomeip-sys/build.rs]: Done

File: .github/workflows/lint-and-test.yaml [hunk 21]
```diff
@@ -28,16 +40,132 @@ concurrency:
       cancel-in-progress: true
 
 jobs:
+  download_and_decompress:
```
[the remark this answers is not in the document]
Comment (PLeVasseur) [reply .github/workflows/lint-and-test.yaml:43]: I'm going to open an issue to look into this more, but mark resolved for now to keep moving.

**Comment by PLeVasseur** [comment]
> I like the idea of being able to build the vsomeip library as part of the transport's build. Have you taken a look at other *-sys crates and how they do this? Maybe https://github.com/eclipse/paho.mqtt.rust/tree/master/paho-mqtt-sys can provide some additional inspiration ...

Seems like the answer is "use a git submodule", much like you did for up-rust to use the up-spec protos.

I'll look into this.

File: .github/workflows/lint-and-test.yaml [hunk 22]
```diff
@@ -63,21 +111,21 @@ jobs:
       - name: Run tests and report code coverage
         run: |
           # enable nightly features so that we can also include Doctests
-          RUSTC_BOOTSTRAP=1 cargo tarpaulin -o xml -o lcov -o html --doc --tests
+          RUSTC_BOOTSTRAP=1 cargo tarpaulin -o xml -o lcov -o html --doc --tests -- --test-threads 1
```
Comment (sophokles73) [inline .github/workflows/lint-and-test.yaml:114]: on my local Ubuntu 22.04 machine I need to explicitly set LD_LIBRARY_PATH to also include `${VSOMEIP_INSTALL_PATH}/lib` for the tests to run successfully ...

File: Cargo.toml [hunk 23]
```diff
@@ -25,22 +31,22 @@ exclude = [
...
-[dependencies]
+[workspace.dependencies]
 async-trait = { version = "0.1" }
 bytes = { version = "1.5" }
 rand = { version = "0.8" }
 regex = { version = "1.10" }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = { version = "1.0" }
-up-rust = { git = "https://github.com/eclipse-uprotocol/up-rust", version = "0.1.5" }
+up-rust = { git = "https://github.com/eclipse-uprotocol/up-rust", rev = "3a50104421a801d52e1d9c68979db54c013ce43d" }
```
Comment (sophokles73) [inline Cargo.toml:46]: is there a reason why you switched back to the commit instead of e.g. using version `0.1`?

**Review by sophokles73 — COMMENTED** [review]
Being able to build vsomeip-sys using the `bundled` feature is great :+1:

File: .github/workflows/lint-and-test.yaml (hunk 21 above)
[the remark this answers is not in the document]
Comment (PLeVasseur) [reply .github/workflows/lint-and-test.yaml]: Okay, I now test for both with the bundled feature flag turned on and off.

File: Cargo.toml (hunk 23 above)
Comment (PLeVasseur) [reply Cargo.toml:46]: Didn't switch. I mentioned on the other PR that I have a separate PR updating to 0.1 over [here](https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/pull/13).

File: .github/workflows/lint-and-test.yaml [hunk 24]
```diff
@@ -28,16 +33,38 @@ concurrency:
...
+
   lint:
     name: Lint
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
+
+      - name: Install C++ dependencies
+        run: sudo apt-get install -y build-essential cmake libboost-all-dev
```
[the remark this answers is not in the document]
Comment (PLeVasseur) [reply .github/workflows/lint-and-test.yaml]: Added this

File: .github/workflows/lint-and-test.yaml (hunk 22 above)
Comment (PLeVasseur) [reply .github/workflows/lint-and-test.yaml]: Added this

File: vsomeip-sys/README.md [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
+Then,
+
+```bash
+VSOMEIP_LIB_PATH=<path/to/vsomeip/lib> GENERIC_CPP_STDLIB_PATH=<path/to/generic/cpp/stdlib> ARCH_SPECIFIC_CPP_STDLIB_PATH=<path/to/arch_specific/cpp/stdlib> cargo build
+```
+
+If you are running a Linux-based OS on x86_64, it's possible these look like:
+
+```
+GENERIC_CPP_STDLIB_PATH = /usr/include/c++/11
```
[the remark this answers is not in the document]
Comment (PLeVasseur) [reply vsomeip-sys/README.md]: Fixed the formatting

**Comment by PLeVasseur** [comment]
> Being able to build vsomeip-sys using the `bundled` feature is great 👍

Hey @sophokles73 -- can you take another look?

I re-added build steps in which we don't use the `bundled` feature to make sure we support that use case.

Also addressed the comments you had left.

Thx!

**Review by sophokles73 — APPROVED** [review]
LGTM
